spec/models/verification/residence_spec.rb

```diff
@@ -93,16 +93,16 @@
 
   describe "Failed census call" do
     it "stores failed census API calls" do
-      residence = build(:verification_residence, :invalid, document_number: "12345678Z")
+      residence = build(:verification_residence, document_number: "87654321Z")
```

Comment:
I wanted to make the CensusApi return the invalid stubbed response directly; otherwise, the failure is not because of an invalid CensusAPI response; it is because the postal_code provided by the user and the one returned on the response are not coincident. Of course, we consider this case a failed response, but I think it will be easier to understand for developers why this test passes doing this way.

In my opinion, to call the factory with the `:invalid` trait that setups a valid postal code is confusing.

We can fix the same spec by changing the `:invalid` trait as follows and removing restoring the original test code:

```
  factory :verification_residence, class: Verification::Residence do
    user
    document_number
    document_type    { "1" }
    date_of_birth    { Time.zone.local(1980, 12, 31).to_date }
    postal_code      { "30813" }
    terms_of_service { "1" }

    trait :invalid do
      postal_code { "30800" }
    end
  end
```

Reply:
I agree the code is confusing :sweat_smile:. However, I'd rather make it confusing in the same way CONSUL's code is confusing than in a different way :thinking:. So, yeah, I'd go for changing the factory :grimacing:.

Reply:
Done